qutip/tests/core/data/test_mathematics.py

```diff
@@ -400,14 +406,16 @@ def test_mathematically_correct(self, op, data_m, out_type):
         if issubclass(out_type, Data):
             assert test.shape == expected.shape
             np.testing.assert_allclose(test.to_array(), expected,
-                                       atol=self.tol)
+                                       atol=self.tol,
+                                       rtol=self.rtol)
         elif out_type is list:
             for test_, expected_ in zip(test, expected):
                 assert test_.shape == expected_.shape
                 np.testing.assert_allclose(test_.to_array(),
-                                           expected_, atol=self.tol)
+                                           expected_, atol=self.tol,
+                                           rtol=self.rtol)
         else:
-            assert abs(test - expected) < self.tol
+            assert abs(test - expected) < self.tol + self.rtol*abs(expected)
```

Comment:
We may want to use `np.testing.assert_allclose` for consistency here.

Reply:
For scalar, we could also use `pytest.approx`.

Reply:
I don't mind too much which we pick, but let's not code this ourselves. I'll put in a small vote for `pytest.approx` (it prints nice error messages) although its behaviour is slightly different (which counts against it a bit in this file).

Reply:
I would vote for numpy just for consistency in the behaviour but I am really fine with both.

Reply:
Sounds good!